### PR TITLE
Typescript declaration file for WebAssembly JS API

### PR DESCRIPTION
**Pull request by winksaville**
Please fill in this template.

- [y] Use a meaningful title for the pull request. Include the name of the package modified.
- [y] Test the change in your own code. (Compile and run.)
- [y] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [y] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [y] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If adding a new definition:
- [y] The package does not provide its own types, and you can not add them.
- [y] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [y] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [y] `tslint.json` should be present, and `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, and `strictNullChecks` set to `true`.

If changing an existing definition:
- [ ] Provide a URL to documentation or source code which provides context for the suggested changes: <<url here>>
- [ ] Increase the version number in the header if appropriate.
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "../tslint.json" }`.

If removing a declaration:
- [ ] If a package was never on DefinitelyTyped, you don't need to do anything. (If you wrote a package and provided types, you don't need to register it with us.)
- [ ] Delete the package's directory.
- [ ] Add it to `notNeededPackages.json`.

### CODE REVIEW

**Comment by winksaville** [comment]
I'll look at this in the morning.

**Comment by andy-ms** [comment]
The header isn't in the right format. Use something like the one in `abs/index.d.ts`, just changing the name.
Also, we won't publish a custom README, so any important comments should go in the definition itself.

File: types/webassembly-js-api/index.d.ts [hunk 1]
```diff
@@ -1,7 +1,31 @@
 // Type definitions for WebAssembly v1 (MVP)
 // Project: https://github.com/winksaville/test-webassembly-js-ts
-// Definitions by: 01alchemist (https://twitter.com/01alchemist) && Wink Saville (https://github.com/winksaville)
+// Definitions by: 01alchemist <https://twitter.com/01alchemist>
```
Comment (andy-ms) [inline types/webassembly-js-api/index.d.ts]: Use `,` instead of `&&`

File: types/webassembly-js-api/index.d.ts [hunk 2]
```diff
@@ -1,7 +1,31 @@
...
 // Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+/**
+ * WebAssembly v1 (MVP) declaration file for TypeScript
```
Comment (andy-ms) [inline types/webassembly-js-api/index.d.ts]: Duplicates info in header.

File: types/webassembly-js-api/index.d.ts [hunk 3]
```diff
@@ -1,7 +1,31 @@
...
+ *
+ * Authors: 01alchemist <https://twitter.com/01alchemist>
+ *          and Wink Saville <wink@saville.com>
+ *
+ * ## Installation
+ * ```
+ * npm install @types/webassembly-js-api --save-dev
```
Comment (andy-ms) [inline types/webassembly-js-api/index.d.ts]: This is the standard installation instruction, no need to duplicate here.

**Comment by winksaville** [comment]
@andy-ms Thanks for the assistance.

File: types/webassembly-js-api/index.d.ts [hunk 4]
```diff
@@ -1,7 +1,31 @@
...
+ * ```
+ *
+ * The WebAssembly namespace will now be available to your code TypeScript code.
+ * See [WebAssembly JS API](http://webassembly.org/getting-started/js-api/) and the contents of
```
Comment (andy-ms) [inline types/webassembly-js-api/index.d.ts]: This should be the `Project` link.

File: types/webassembly-js-api/index.d.ts [hunk 5]
```diff
@@ -1,7 +1,31 @@
...
+ * @types/webassembly-js-api/index.d.ts for more details.
+ *
+ * Do not use an `import * as WebAssembly from "webassembly-js-api"` as this will cause
```
Comment (andy-ms) [inline types/webassembly-js-api/index.d.ts]: That will be a compiler error anyway.

File: types/webassembly-js-api/index.d.ts [hunk 6]
```diff
@@ -1,7 +1,31 @@
...
+ * the `tsc` to emit a `require` statement for the WebAssembly module which doesn't exist as
+ * WebAssembly is part of the global namespace.
+ *
+ * Instead use a `///<reference path="./node_modules/@types/webassembly-js-api"/>` statement.
```
Comment (andy-ms) [inline types/webassembly-js-api/index.d.ts]: We recommend `/// <reference types>` instead, so no need for a comment here.

**Comment by winksaville** [comment]
Streamlined that second comment block:
```
/**
 * The WebAssembly namespace, see [WebAssembly](https://github.com/webassembly)
 * and [WebAssembly JS API](http://webassembly.org/getting-started/js-api/)
 * for more information.
 */
```

**Comment by winksaville** [comment]
Is there anything more that needs to be done before this can be merged?

**Comment by winksaville** [comment]
@RyanCavanaugh & @andy-ms thanks for merging and the reviews.

**Pull request merged**